spotty/providers/aws/cfn_templates/instance/template.py

```diff
@@ -56,20 +56,16 @@ def prepare_instance_template(ec2, instance_config: InstanceConfig, docker_comma
             }]
         del template['Resources']['InstanceLaunchTemplate']['Properties']['LaunchTemplateData']['SecurityGroupIds']
 
-    # add ports to the security group
-    for port in instance_config.ports:
-        if port != 22:
-            template['Resources']['InstanceSecurityGroup']['Properties']['SecurityGroupIngress'] += [{
-                'CidrIp': '0.0.0.0/0',
-                'IpProtocol': 'tcp',
-                'FromPort': port,
-                'ToPort': port,
-            }, {
-                'CidrIpv6': '::/0',
-                'IpProtocol': 'tcp',
-                'FromPort': port,
-                'ToPort': port,
-            }]
+    # add ports to the security group for the user-defined inbound IP
+    inbound_ip = instance_config.inbound_ip
+    cidr_ip = 'CidrIpv6' if inbound_ip == 6 else 'CidrIp'
```

Comment:
It looks like you forgot to add a proper check for IPv6 addresses here

Reply:
woops! This what happens when I don't do the unit tests - maybe the templates could to with some tests to ensure the args are at least building them correctly...